src/Record.js

```diff
@@ -120,7 +120,11 @@ export class Record {
 
   equals(other) {
     return (
-      this === other || (other && recordSeq(this).equals(recordSeq(other)))
+      this === other ||
+      (other &&
+        this._keys &&
+        other._keys &&
```

Comment:
I wonder if the proper way to do this should be `isRecord(other)`?

Reply:
It's definitely semantically more obvious what the check is doing but it depends on the guarantee that Record will always have _keys instantiated, which we will be alerted on if that is ever violated with this newly added unit test.